### PR TITLE
talecast: init at 0.1.39

### PR DESCRIPTION
**Pull request by confusedalex**
## Description of changes
https://github.com/TBS1996/TaleCast
Simple CLI podcatcher

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ta/talecast/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  meta = {
+    description = "Simple CLI podcatcher";
+    homepage = "https://github.com/TBS1996/TaleCast";
+    license = lib.licenses.mit;
+    mainProgram = "talecast";
+    maintainers = [ lib.maintainers.confusedalex ];
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:30]:
```suggestion
    maintainers = with lib.maintainers; [ confusedalex getchoo ];
```

this allows for cleaner diffs when new maintainers are added. feel free to add me btw :)

File: pkgs/by-name/ta/talecast/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:6]:
```suggestion
  openssl,
  nix-update-script,
  testers,
  talecast
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    repo = pname;
+    rev = "816e11dcc5eafbe54d2ecd4bc525a46f88a3e154";
+    hash = "sha256-sR0wyzAwTTwO+HHly4I0JJ5HwxJUD1yGdKl9LqEbnI0=";
+  };
+
+  cargoHash = "sha256-mIzrYlAqHYrK2bb/ZUzqIwhPJKcTQpNpqijpEuwLc5A=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:24]:
```suggestion
  passthru = {
    updateScript = nix-update-script { };
    tests.version = testers.testVersion { package = talecast; };
  };

```
we can adding basic tests and an update script here to make this easier to maintain

**Review by getchoo — CHANGES_REQUESTED** [review]
welcome to nixpkgs! just a few notes

File: pkgs/by-name/ta/talecast/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "talecast";
+  version = "0.1.39";
```
[the remark this answers is not in the document]
Comment (ConfusedAlex) [reply pkgs/by-name/ta/talecast/package.nix:11]: I'm really glad for your suggestions and the warm welcoming :) The latest tag is 1.20, but the cargo package versions says 1.39. Which versions should be used?

[the remark this answers is not in the document]
Comment (getchoo) [reply pkgs/by-name/ta/talecast/package.nix:11]: it seems 1.39 is used on [crates.io](https://crates.io/crates/talecast) as well, so yes i would use that as the tag instead in this case

File: pkgs/by-name/ta/talecast/package.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  fetchFromGitHub,
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:3]:
```suggestion
  fetchCrate,
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  pkg-config,
+  openssl,
+  nix-update-script,
+  testers,
+  talecast
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "talecast";
+  version = "0.1.39-unstable-2024-05-21";
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:14]:
```suggestion
  version = "0.1.39";
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  src = fetchFromGitHub {
+    owner = "TBS1996";
+    repo = "talecast";
+    rev = "816e11dcc5eafbe54d2ecd4bc525a46f88a3e154";
+    hash = "sha256-sR0wyzAwTTwO+HHly4I0JJ5HwxJUD1yGdKl9LqEbnI0=";
+  };
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:21]:
```suggestion
  src = fetchCrate {
    inherit pname version;
    hash = "sha256-RwB+X+i3CEcTyKac81he9/cT2aQ4M7AqgqSDBEvhFJU=";
  };
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  cargoHash = "sha256-39WmmRbxeiAnZoUR1OibzklORxauG2L8PYBx1m3Uibc=";
```
Comment (getchoo) [inline pkgs/by-name/ta/talecast/package.nix:23]:
```suggestion
  cargoHash = "sha256-mIzrYlAqHYrK2bb/ZUzqIwhPJKcTQpNpqijpEuwLc5A=";
```

**Review by getchoo — CHANGES_REQUESTED** [review]
just thought of this: we can fetch from crates.io here instead of github to target the actual `0.1.39`. this makes sure `passthru.updateScript` works as intended and also fixes `tests.version` (as it's failing right now due to including the `-unstable` version suffix)

**Review by getchoo — APPROVED** [review]
LGTM. thanks!

Result of `nixpkgs-review pr 321113` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>talecast</li>
  </ul>
</details>

**Comment by h7x4** [comment]
I believe the commits are ordered the wrong way?

**Comment by ConfusedAlex** [comment]
> I believe the commits are ordered the wrong way?

fixed now :)

**Review by getchoo — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/ta/talecast/package.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-RwB+X+i3CEcTyKac81he9/cT2aQ4M7AqgqSDBEvhFJU=";
+  };
+
+  cargoHash = "sha256-mIzrYlAqHYrK2bb/ZUzqIwhPJKcTQpNpqijpEuwLc5A=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/talecast/package.nix:24]:
```suggestion

  buildInputs = [ openssl ]
    + lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 10]
```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  fetchCrate,
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/talecast/package.nix:3]:
```suggestion
  fetchCrate,
  darwin,
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  darwin,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  nix-update-script,
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/talecast/package.nix:8]:
```suggestion
  nix-update-script,
  stdenv,
```

File: pkgs/by-name/ta/talecast/package.nix [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    inherit pname version;
+    hash = "sha256-RwB+X+i3CEcTyKac81he9/cT2aQ4M7AqgqSDBEvhFJU=";
+  };
+
+  cargoHash = "sha256-mIzrYlAqHYrK2bb/ZUzqIwhPJKcTQpNpqijpEuwLc5A=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ]
+    + lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/ta/talecast/package.nix:28]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```

**Comment by SuperSandro2000** [comment]
@NixOS/darwin-maintainers can you help here? ofborg is taking it's sweet sweet time again

**Comment by afh** [comment]
Result of `nixpkgs-review pr 321113` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>talecast</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Nice, if that was all we can just merge it 😂

**Review by afh — APPROVED** [review]
LGTM 👍

@SuperSandro2000 Would it be possible to squash the additional commits into the talecast init one?

**Comment by SuperSandro2000** [comment]
I just squash merged. I cannot force push the PR because it was made from the master branch.

**Pull request merged**